CanHazFunny/CanHazFunny/Jester.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Diagnostics.Contracts;
+
+namespace CanHazFunny
+{
+    public class Jester
+    {
+        private JokeService _jokeService;
+        private JokeWriter _jokeWriter;
```

Comment:
The point of using interfaces is to add a layer of abstraction between the depending class and the dependency. If we set the types of these two members to a concrete class, we would have to update this code if we wanted to use different classes that implemented the same interfaces.
```suggestion
        private IGetJoke_jokeService;
        private IWriteJoke_jokeWriter;
```